src/neo/Network/P2P/ProtocolHandler.cs

```diff
@@ -148,6 +151,19 @@ private void OnMessage(Message msg)
             }
         }
 
+        private void OnOracleReceived(OraclePayload payload)
+        {
+            // Verify
+
+            using var snapshot = Blockchain.Singleton.GetSnapshot();
+            if (payload.Verify(snapshot)) return;
+
```

Comment:
Should we check whether the `OraclePayload` has a valid oracle request? Is it possible, one oracle node send a lot of useless `OraclePayload` to attack the network?

Reply:
It's possible but it will be enqueued and discarded with time, but I will order it for count, and grouping by publicKey

Reply:
It was improved the pool's sort in this order:

- Sort by number of signatures
- Sort by if has my signature or not
- Sort by age

Reply:
Also we can add a PoW in order to prevent the p2p spam, but I think that it's not required